File: Java/DesignPatterns/src/test/java/designPatterns/structural/proxy/MainProxyTest.java
package designPatterns.structural.proxy;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainProxyTest {
  private Subject subject;

  @BeforeEach
  void setUp() {
    subject = new Proxy("proxyValue");
  }

  @Test
  void equalsTest() {
    assertThat(subject.display()).isEqualTo("Displaying proxyValue");
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/structural/proxy/MainProxyTest.kt
package structural.proxy

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MainProxyTest {
  private lateinit var subject: Subject

  @BeforeEach
  internal fun setUp() {
    subject = Proxy("proxyValue")
  }

  @Test
  internal fun equalsTest() {
    assertThat(subject.display()).isEqualTo("Displaying proxyValue")
  }
}